src/run/run.c

```diff
@@ -2056,18 +2095,9 @@ static int match_job_removed(sd_bus_message *m, void *userdata, sd_bus_error *er
         if (!streq_ptr(path, c->start_job))
                 return 0;
 
-        /* Notify our caller that the service is now running, just in case. */
-        (void) sd_notifyf(/* unset_environment= */ false,
-                          "READY=1\n"
-                          "RUN_UNIT=%s",
-                          c->unit);
-
-        r = run_context_setup_ptyfwd(c);
+        r = run_context_check_started(c, /* current_job = */ NULL);
```

Comment:
(I think the match for JobRemoved can be removed wholesale. Now that active_state is checked in _started() we might be looking at outdated data here)